Generate serialization for ScrollingStateFrameHostingNode

achristensen07 opened the pull request:
#### 887ac88a0cce09a36aa033be0cef3998f9067ff2
<pre>
Generate serialization for ScrollingStateFrameHostingNode
<a href="https://bugs.webkit.org/show_bug.cgi?id=263704">https://bugs.webkit.org/show_bug.cgi?id=263704</a>
rdar://117513008

Reviewed by Brady Eidson.

This required some new constructors, but the interesting part was teaching the
code generator how to handle types with OptionalTupleBits and non-OptionalTuple members.

* Source/WebCore/page/scrolling/ScrollingStateFrameHostingNode.cpp:
(WebCore::ScrollingStateFrameHostingNode::create):
(WebCore::ScrollingStateFrameHostingNode::ScrollingStateFrameHostingNode):
* Source/WebCore/page/scrolling/ScrollingStateFrameHostingNode.h:
* Source/WebCore/page/scrolling/ScrollingStateNode.cpp:
(WebCore::ScrollingStateNode::ScrollingStateNode):
* Source/WebCore/page/scrolling/ScrollingStateNode.h:
* Source/WebKit/Scripts/generate-serializers.py:
(SerializedType.has_optional_tuple_bits):
(check_type_members):
(encode_type):
(decode_type):
(generate_one_impl):
(generate_serialized_type_info):
(encode_optional_tuple): Deleted.
(decode_optional_tuple): Deleted.
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.cpp:
(IPC::ArgumentCoder&lt;Namespace::Subnamespace::StructName&gt;::encode):
(IPC::ArgumentCoder&lt;Namespace::OtherClass&gt;::encode):
(IPC::ArgumentCoder&lt;Namespace::ReturnRefClass&gt;::encode):
(IPC::ArgumentCoder&lt;Namespace::EmptyConstructorStruct&gt;::encode):
(IPC::ArgumentCoder&lt;Namespace::EmptyConstructorWithIf&gt;::encode):
(IPC::ArgumentCoder&lt;WithoutNamespace&gt;::encode):
(IPC::ArgumentCoder&lt;WithoutNamespaceWithAttributes&gt;::encode):
(IPC::ArgumentCoder&lt;WebCore::InheritsFrom&gt;::encode):
(IPC::ArgumentCoder&lt;WebCore::InheritanceGrandchild&gt;::encode):
(IPC::ArgumentCoder&lt;WTF::Seconds&gt;::encode):
(IPC::ArgumentCoder&lt;WTF::CreateUsingClass&gt;::encode):
(IPC::ArgumentCoder&lt;WebCore::FloatBoxExtent&gt;::encode):
(IPC::ArgumentCoder&lt;SoftLinkedMember&gt;::encode):
(IPC::ArgumentCoder&lt;Namespace::ConditionalCommonClass&gt;::encode):
(IPC::ArgumentCoder&lt;Namespace::CommonClass&gt;::encode):
(IPC::ArgumentCoder&lt;Namespace::AnotherCommonClass&gt;::encode):
(IPC::ArgumentCoder&lt;WebCore::MoveOnlyDerivedClass&gt;::encode):
(IPC::ArgumentCoder&lt;WebKit::LayerProperties&gt;::encode):
(IPC::ArgumentCoder&lt;WebKit::LayerProperties&gt;::decode):
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;WebKit::Fabulous&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;WebCore::Amazing&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;JSC::Incredible&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;WebKit::TemplateTest&lt;Testing::StorageSize&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;WebCore::ScrollingStateFrameHostingNode&gt;::encode):
(IPC::ArgumentCoder&lt;WebCore::ScrollingStateFrameHostingNode&gt;::decode):
(IPC::ArgumentCoder&lt;WebCore::ScrollingStateFrameHostingNodeWithStuffAfterTuple&gt;::encode):
(IPC::ArgumentCoder&lt;WebCore::ScrollingStateFrameHostingNodeWithStuffAfterTuple&gt;::decode):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h:
* Source/WebKit/Scripts/webkit/tests/SerializedTypeInfo.cpp:
(WebKit::allSerializedTypes):
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in:
* Source/WebKit/Scripts/webkit/tests/WebKitPlatformGeneratedSerializers.cpp:
(IPC::ArgumentCoder&lt;WebKit::PlatformClass&gt;::encode):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.cpp:
(ArgumentCoder&lt;WebCore::ScrollingStateFrameHostingNode&gt;::encode): Deleted.
(ArgumentCoder&lt;WebCore::ScrollingStateFrameHostingNode&gt;::decode): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.serialization.in:

Canonical link: <a href="https://commits.webkit.org/269811@main">https://commits.webkit.org/269811@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c797d7ec9637ee5221d14262699a60921734748a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23622 "5 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1738 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24743 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/25781 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21806 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/23890 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3311 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24158 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22374 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23865 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1284 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20438 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26377 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/23805 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1076 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21344 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27621 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21549 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21613 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25382 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1051 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18756 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1029 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1464 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3030 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1340 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

beidson commented on Source/WebKit/Scripts/webkit/tests/WebKitPlatformGeneratedSerializers.cpp:
```diff
@@ -69,6 +69,7 @@ void ArgumentCoder<WebKit::PlatformClass>::encode(Encoder& encoder, const WebKit
     static_assert(MembersInCorrectOrder < 0
         , offsetof(WebKit::PlatformClass, value)
     >::value);
+
```
Stray?

beidson reviewed (approved):
LGTM.

I don't have the skillset to be hypercritical of the python changes, but they pass the sniff test and are regression tested, so...  👍

achristensen07 committed:
Generate serialization for ScrollingStateFrameHostingNode
https://bugs.webkit.org/show_bug.cgi?id=263704
rdar://117513008

Reviewed by Brady Eidson.

This required some new constructors, but the interesting part was teaching the
code generator how to handle types with OptionalTupleBits and non-OptionalTuple members.

* Source/WebCore/page/scrolling/ScrollingStateFrameHostingNode.cpp:
(WebCore::ScrollingStateFrameHostingNode::create):
(WebCore::ScrollingStateFrameHostingNode::ScrollingStateFrameHostingNode):
* Source/WebCore/page/scrolling/ScrollingStateFrameHostingNode.h:
* Source/WebCore/page/scrolling/ScrollingStateNode.cpp:
(WebCore::ScrollingStateNode::ScrollingStateNode):
* Source/WebCore/page/scrolling/ScrollingStateNode.h:
* Source/WebKit/Scripts/generate-serializers.py:
(SerializedType.has_optional_tuple_bits):
(check_type_members):
(encode_type):
(decode_type):
(generate_one_impl):
(generate_serialized_type_info):
(encode_optional_tuple): Deleted.
(decode_optional_tuple): Deleted.
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.cpp:
(IPC::ArgumentCoder<Namespace::Subnamespace::StructName>::encode):
(IPC::ArgumentCoder<Namespace::OtherClass>::encode):
(IPC::ArgumentCoder<Namespace::ReturnRefClass>::encode):
(IPC::ArgumentCoder<Namespace::EmptyConstructorStruct>::encode):
(IPC::ArgumentCoder<Namespace::EmptyConstructorWithIf>::encode):
(IPC::ArgumentCoder<WithoutNamespace>::encode):
(IPC::ArgumentCoder<WithoutNamespaceWithAttributes>::encode):
(IPC::ArgumentCoder<WebCore::InheritsFrom>::encode):
(IPC::ArgumentCoder<WebCore::InheritanceGrandchild>::encode):
(IPC::ArgumentCoder<WTF::Seconds>::encode):
(IPC::ArgumentCoder<WTF::CreateUsingClass>::encode):
(IPC::ArgumentCoder<WebCore::FloatBoxExtent>::encode):
(IPC::ArgumentCoder<SoftLinkedMember>::encode):
(IPC::ArgumentCoder<Namespace::ConditionalCommonClass>::encode):
(IPC::ArgumentCoder<Namespace::CommonClass>::encode):
(IPC::ArgumentCoder<Namespace::AnotherCommonClass>::encode):
(IPC::ArgumentCoder<WebCore::MoveOnlyDerivedClass>::encode):
(IPC::ArgumentCoder<WebKit::LayerProperties>::encode):
(IPC::ArgumentCoder<WebKit::LayerProperties>::decode):
(IPC::ArgumentCoder<WebKit::TemplateTest<WebKit::Fabulous>>::encode):
(IPC::ArgumentCoder<WebKit::TemplateTest<WebCore::Amazing>>::encode):
(IPC::ArgumentCoder<WebKit::TemplateTest<JSC::Incredible>>::encode):
(IPC::ArgumentCoder<WebKit::TemplateTest<Testing::StorageSize>>::encode):
(IPC::ArgumentCoder<WebCore::ScrollingStateFrameHostingNode>::encode):
(IPC::ArgumentCoder<WebCore::ScrollingStateFrameHostingNode>::decode):
(IPC::ArgumentCoder<WebCore::ScrollingStateFrameHostingNodeWithStuffAfterTuple>::encode):
(IPC::ArgumentCoder<WebCore::ScrollingStateFrameHostingNodeWithStuffAfterTuple>::decode):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h:
* Source/WebKit/Scripts/webkit/tests/SerializedTypeInfo.cpp:
(WebKit::allSerializedTypes):
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in:
* Source/WebKit/Scripts/webkit/tests/WebKitPlatformGeneratedSerializers.cpp:
(IPC::ArgumentCoder<WebKit::PlatformClass>::encode):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTree.serialization.in:
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.cpp:
(ArgumentCoder<WebCore::ScrollingStateFrameHostingNode>::encode): Deleted.
(ArgumentCoder<WebCore::ScrollingStateFrameHostingNode>::decode): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.serialization.in:

Canonical link: https://commits.webkit.org/269811@main
